### PR TITLE
fix: asset search auto focus

### PR DESCRIPTION
**Pull request by chxash**
## Description

This enables auto focus on asset search input box for trade asset / search asset / buy and sell asset modals

## Notice

- [X] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [X] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

https://github.com/shapeshift/web/issues/1723

## Risk

No particular risk identified

## Testing

- Navigate to assets page
- Verify input field is auto focused
- Click on one asset in the trade window
- Verify input field is auto focused
- Click on buy / sell crypto => Gem / Buy => Select an asset to buy
- Verify input field is auto focused

## Screenshots (if applicable)

assetSearch1

assetbuy

trade

### CODE REVIEW

File: src/components/AssetSearch/AssetSearch.tsx [hunk 1]
```diff
@@ -51,6 +51,7 @@ export const AssetSearch = ({ onClick, filterBy }: AssetSearchProps) => {
             {...register('search')}
             type='text'
             placeholder='Search'
+            autoFocus // eslint-disable-line jsx-a11y/no-autofocus
```
Comment (NeOMakinG) [inline src/components/AssetSearch/AssetSearch.tsx:54]: If we use autofocus, why do we have this rule activated?

Comment (chxash) [reply src/components/AssetSearch/AssetSearch.tsx:54]: good question !

**Review by gomesalexandre — DISMISSED** [review]
Ran locally, input for gem modal, trade component, and assets page is fixed 🍬

**Comment by chxash** [comment]
> @NeOMakinG @chxash if either of you want to make an issue/PR that
> 
> * removes the lint rule
> * removes existing comments that disable the lint rule for housekeeping
> 
> i'll gladly approve and merge as per @reallybeard
> 
> as far as this PR goes, LGTM

I will do that !

**Comment by 0xdef1cafe** [comment]
@chxash we've had flaky cypress tests recently, but i've rerun this multiple times and it seems to fail in the same place.

can you please check with `yarn test:cypress` or `yarn test:cypress:headless` locally to reproduce, debug the issue, and update the cypress test case?

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]
please see comment - i believe this is causing cypress tests to fail

**Comment by chxash** [comment]
> please see comment - i believe this is causing cypress tests to fail

I have ran cypress tests multiple times, and I had the same failure at first try, then they passed everytime.

It's the same issue as always, following line is timing out waiting for requests in `dashboard_spec.ts`: 
`cy.get('[data-test=trade-form-preview-button]').contains('Insufficient Funds', {
      timeout: 60000,
})`

I could increase the timeout again ? but I am not sure

**Comment by 0xdef1cafe** [comment]
@chxash confirmed - apologies, we did think this was related to trade tests failing as this component is used by the swapper, but indeed it did just pass CI when i ran it again now.

**Comment by chxash** [comment]
> @chxash confirmed - apologies, we did think this was related to trade tests failing as this component is used by the swapper, but indeed it did just pass CI when i ran it again now.

@0xdef1cafe  no need for apologies really ! I was doubting as well.

**Pull request merged**